src/Bicep.Core/SemanticModel/NameBindingVisitor.cs

```diff
@@ -110,30 +131,61 @@ private Symbol ValidateFunctionFlags(Symbol symbol, TextSpan span)
             return symbol;
         }
 
-        private Symbol LookupSymbolByName(string name, TextSpan span)
+        private Symbol LookupSymbolByName(string name, TextSpan span, string? @namespace)
         {
-            if (this.declarations.TryGetValue(name, out var localSymbol))
+            NamespaceSymbol? FindNamespace(string name) =>
+                this.namespaces
+                .Where(ns => ns.Name.Equals(name, LanguageConstants.IdentifierComparison))
+                .FirstOrDefault();
+
+            Symbol? foundSymbol;
+            if (string.IsNullOrEmpty(@namespace))
             {
-                // we found the symbol in the local namespace
-                return ValidateFunctionFlags(localSymbol, span);
+                // attempt to find name in the imported namespaces
+                var namespaceSymbol = FindNamespace(name);
+
+                if (namespaceSymbol != null)
+                {
+                    // namespace symbol found
+                    return ValidateFunctionFlags(namespaceSymbol, span);
+                }
+
+                // declarations must not have a namespace value, namespaces are used to fully qualify a function access.
+                // There might be instances where a variable declaration for example uses the same name as one of the imported
+                // functions, in this case to differentiate a variable declaration vs a function access we check the namespace value,
+                // the former case must have an empty namespace value whereas the latter will have a namespace value.
+                if (this.declarations.TryGetValue(name, out var localSymbol))
+                {
+                    // we found the symbol in the local namespace
+                    return ValidateFunctionFlags(localSymbol, span);
+                }
+
+                // attempt to find function in all imported namespaces
+                var foundSymbols = this.namespaces
+                    .Select(ns => ns.TryGetFunctionSymbol(name))
+                    .Where(symbol => symbol != null)
+                    .ToList();
+
+                if (foundSymbols.Count() > 1)
```

Comment:
This isn't user-actionable as there isn't any logic to resolve symbols across multiple namespaces. Should we just enforce uniqueness when importing namespaces so that we can write tests to avoid collisions?

Reply:
What do you mean? This is not possible right now because the two namespaces don't have functions with the same names. If that ever happened, the user action is to fully qualify the function name with a namespace. (Before @jorgecotillo's changes, this code was not reachable.)

I'm fine with asserting in tests that the two built-in namespaces don't overlap, but we should not remove this check.

Reply:
Ah, I misunderstood. Yeah that makes total sense.